docs/user-guide/grub.rst

```diff
@@ -0,0 +1,155 @@
+the directory where it has been loaded from (grub subdirectory in our case). It is known for (specific versions?) KVM
+and ppc64le that grub may end up loading this as first grub.cfg. We simply set ``prefix="grub"`` and manually load the
+main config file ``grub/grub.cfg``.
+
+``grub/grub.cfg``
++++++++++++++++++
+
+This is the main entry point for all architectures. We always load this config file.
+
+``grub/grub/local_*.cfg``
++++++++++++++++++++++++++
+
+This are the architecture specific config files providing local (hard disk) boot entries. These may need adjusting over
+the time, depending how distributions name their ``*.efi`` executable for local boot
+
+``grub/grub/system/*``
+++++++++++++++++++++++
+
+Empty directory where cobbler will sync machine specific configuration (typically setting local boot or an
+(auto-)install menu entry). These are named after the mac address of a machine, e.g.: ``grub/system/52:54:00:42:07:04``
+This config file is tried to be loaded from the main ``grub.cfg``.
+
+``grub/grub/system_link.*``
++++++++++++++++++++++++++++
+
+Empty directory where cobbler will create symlinks, named after the cobbler name of the machine and it links to above
+described mac address file in ``../system/${mac}`` This is only for easier reading and debugging of machine specific
+grub settings.
+
+
+/var/lib/cobbler/loaders
+========================
+
+This directory holds network bootloaders (or links to them) and is also synced to ``/srv/tftp`` root directory 1 to 1.
+Its contents is created in the post (removed in postun) section of the ``cobbler.spec`` file. This is done via
+``scripts/mkgrub.sh`` shell script.
+
+It creates grub executables for each installed grub2-$arch via: ``grub2-mkimage`` and links in the corresponding grub2
+modules and other supported bootloaders (``pxelinux.0``, ...)
+
+If you have installed e.g. a new grub or syslinux version, you should re-run ``mkgrub.sh`` to build new grub
+executables. For other, static or already compiled/linked bootloaders like, shim, ``pxelinux.0`` or a precompiled,
+signed ``grub.efi`` executable, it is enough to call ``cobbler sync`` now (we store links to these now).
+
+The grub specific files generated/linked via ``mkgrub.sh`` are also described here:
+
+``.cobbler_postun_cleanup``
++++++++++++++++++++++++++++
+
+Filled up with generated ``grub2-mkimage`` binaries and created links.
+
+This is needed in ``postun`` ``cobbler.spec`` section to remove things again. This is the only, not synced file.
+
+``grub/grub.0``
++++++++++++++++
+
+- 32 bit PXE (x86 legacy) grub executable.
+- ``grub2-mkimage`` generated.
+- This can/should be used instead of ``pxelinux.0``. You then get the full grub boot process.
+- The bootloader is named ``grub.0``, because ``pxelinux.0`` can chain boot this grub executable via network. But it
+  (or specific versions?) wants boot loaders with a filename ending on ``.0``.
```

Comment:
```suggestion
- 32 bit PXE (x86 legacy) GRUB executable.
- ``grub2-mkimage`` generated.
- This can/should be used instead of ``pxelinux.0``. You then get the full grub boot process.
- The bootloader is named ``grub.0``, because ``pxelinux.0`` can chain boot this grub executable via network. But it
  (or specific versions?) wants bootloaders with a filename ending on ``.0``.
```